pkg/collector/corechecks/containers/containerd.go

```diff
@@ -0,0 +1,359 @@
+		sender.Gauge("containerd.hugetlb.failcount", float64(h.Failcnt), "", tags)
+		sender.Gauge("containerd.hugetlb.usage", float64(h.Usage), "", tags)
+	}
+}
+
+func computeMem(sender aggregator.Sender, mem *cgroups.MemoryStat, tags []string) {
+	memList := map[string]*cgroups.MemoryEntry{
+		"containerd.mem.current":    mem.Usage,
+		"containerd.mem.kernel_tcp": mem.KernelTCP,
+		"containerd.mem.kernel":     mem.Kernel,
+		"containerd.mem.swap":       mem.Swap,
+	}
+	for metricName, memStat := range memList {
+		parseAndSubmitMem(metricName, sender, memStat, tags)
+	}
+	sender.Gauge("containerd.mem.cache", float64(mem.Cache), "", tags)
+	sender.Gauge("containerd.mem.rss", float64(mem.RSS), "", tags)
+	sender.Gauge("containerd.mem.rsshuge", float64(mem.RSSHuge), "", tags)
+	sender.Gauge("containerd.mem.usage", float64(mem.Usage.Usage), "", tags)
+	sender.Gauge("containerd.mem.kernel.usage", float64(mem.Kernel.Usage), "", tags)
+	sender.Gauge("containerd.mem.dirty", float64(mem.Dirty), "", tags)
+}
+
+func parseAndSubmitMem(metricName string, sender aggregator.Sender, stat *cgroups.MemoryEntry, tags []string) {
+	if stat.Size() == 0 {
+		return
+	}
+	sender.Gauge(fmt.Sprintf("%s.usage", metricName), float64(stat.Usage), "", tags)
+	sender.Gauge(fmt.Sprintf("%s.failcnt", metricName), float64(stat.Failcnt), "", tags)
+	sender.Gauge(fmt.Sprintf("%s.limit", metricName), float64(stat.Limit), "", tags)
+	sender.Gauge(fmt.Sprintf("%s.max", metricName), float64(stat.Max), "", tags)
+
+}
+
+func computeCPU(sender aggregator.Sender, cpu *cgroups.CPUStat, tags []string) {
+	sender.Gauge("containerd.cpu.system", float64(cpu.Usage.Kernel), "", tags)
+	sender.Gauge("containerd.cpu.total", float64(cpu.Usage.Total), "", tags)
+	sender.Gauge("containerd.cpu.user", float64(cpu.Usage.User), "", tags)
+	sender.Gauge("containerd.cpu.throttle.periods", float64(cpu.Throttling.Periods), "", tags)
+}
+
+func computeBlkio(sender aggregator.Sender, blkio *cgroups.BlkIOStat, tags []string) {
+	blkioList := map[string][]*cgroups.BlkIOEntry{
+		"containerd.blkio.merged_recursive":        blkio.IoMergedRecursive,
+		"containerd.blkio.queued_recursive":        blkio.IoQueuedRecursive,
+		"containerd.blkio.sectors_recursive":       blkio.SectorsRecursive,
+		"containerd.blkio.service_recursive_bytes": blkio.IoServiceBytesRecursive,
+		"containerd.blkio.time_recursive":          blkio.IoTimeRecursive,
+		"containerd.blkio.serviced_recursive":      blkio.IoServicedRecursive,
+		"containerd.blkio.wait_time_recursive":     blkio.IoWaitTimeRecursive,
+		"containerd.blkio.service_time_recursive":  blkio.IoServiceTimeRecursive,
+	}
+	for metricName, ioStats := range blkioList {
+		parseAndSubmitBlkio(metricName, sender, ioStats, tags)
+	}
+}
+
+func parseAndSubmitBlkio(metricName string, sender aggregator.Sender, list []*cgroups.BlkIOEntry, tags []string) {
+	for _, m := range list {
+		if m.Size() == 0 {
+			continue
+		}
+		blkiotags := []string{
+			fmt.Sprintf("dev:%s", m.Device),
```

Comment:
Is there a reason you are not using the classic `device` tag name?

Reply:
As it's a reserved tag https://docs.datadoghq.com/tagging I was not sure if that might backfire.
Happy to move to `device` if you think it's not a legit concern

Reply:
Indeed, but we are spot on in the scope of this tag (disk metric per device). This is what we're doing for the disk, network, docker and kube checks